bundles/org.openhab.binding.nibeuplink/src/main/java/org/openhab/binding/nibeuplink/internal/connector/UplinkWebInterface.java

```diff
@@ -126,41 +117,91 @@ void enqueue(NibeUplinkCommand command) {
-                        }
-                    }
-                };
+        /**
+         * executes the next command in the queue. requires authenticated session.
+         *
+         * @throws ValidationException
+         */
+        private void executeCommand() {
+            NibeUplinkCommand command = commandQueue.poll();
+            if (command != null) {
+                command.setListener(this::processExecutionResult);
+                command.performAction(httpClient);
+            }
+        }
 
-                NibeUplinkCommand command = commandQueue.poll();
-                if (command != null) {
-                    command.setListener(statusUpdater);
-                    command.performAction(httpClient);
-                }
+        /**
+         * callback that handles result from command execution.
+         *
+         * @param status status information to be evaluated
+         */
+        private void processExecutionResult(CommunicationStatus status) {
+            switch (status.getHttpCode()) {
+                case SERVICE_UNAVAILABLE:
+                    uplinkHandler.setStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                            status.getMessage());
+                    setAuthenticated(false);
+                    break;
+                case FOUND:
+                    uplinkHandler.setStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "most likely your NibeId is wrong. please check your NibeId.");
+                    setAuthenticated(false);
+                    break;
+                case OK:
+                    // no action needed as the thing is already online.
+                    break;
+                default:
+                    uplinkHandler.setStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            status.getMessage());
+                    setAuthenticated(false);
+            }
+        }
+
+        /**
+         * authenticates with the Nibe Uplink WEB interface
+         */
+        private synchronized void authenticate() {
+            setAuthenticated(false);
+            new Login(uplinkHandler, this::processAuthenticationResult).performAction(httpClient);
+        }
+
+        /**
+         * callback that handles result from authentication.
+         *
+         * @param status status information to be evaluated
+         */
+        private void processAuthenticationResult(CommunicationStatus status) {
+            switch (status.getHttpCode()) {
+                case FOUND:
+                    uplinkHandler.setStatusInfo(ThingStatus.ONLINE, ThingStatusDetail.NONE, "logged in");
+                    setAuthenticated(true);
+                    break;
+                case OK:
+                    uplinkHandler.setStatusInfo(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "invalid username or password");
```

Comment:
Not 100% sure if it is mandatory, but i have seen some more PR's where is asked to supply these kind of string as i18n strings. (also check the other occurences, e.g. line 159 etc.)

Reply:
fixed